src/views/PositionPlot/PositionPlotView.tsx

```diff
@@ -0,0 +1,147 @@
+    }, [data.timestamps])
+
+    // Compute the per-panel pixel drawing area dimensions.
+    const panelCount = 1
+    const { panelWidth, panelHeight } = useMemo(() => computePanelDimensions(width, height, panelCount, panelSpacing, margins), [width, height, panelCount])
+    const pixelsPerSecond = useMemo(() => computePixelsPerSecond(panelWidth, startTimeSec, endTimeSec), [endTimeSec, startTimeSec, panelWidth])
+
+    // We need to have the panelHeight before we can use it in the paint function.
+    // By using a callback, we avoid having to complicate the props passed to the painting function; it doesn't make a big difference
+    // but simplifies the prop list a bit.
+    const paintPanel = useCallback((context: CanvasRenderingContext2D, props: PanelProps) => {
+        context.strokeStyle = 'black'
+        context.setLineDash([5, 15]);
+        context.beginPath()
+        context.moveTo(0, props.pixelZero)
+        context.lineTo(width, props.pixelZero)
+        context.stroke()
+        context.setLineDash([]);
+
+        for (let dim of props.dimensions) {
+            context.strokeStyle = colorForUnitId(dim.dimensionIndex)
+            context.beginPath()
+            let first = true
+            for (let i=0; i<dim.pixelTimes.length; i++) {
+                const x = dim.pixelTimes[i]
+                const y = dim.pixelValues[i]
+                if (first) context.moveTo(x, y)
+                else context.lineTo(x, y)
+                first = false
+            }
+            context.stroke()
+        }
+    }, [width])
+
+    // Here we convert the native (time-based spike registry) data to pixel dimensions based on the per-panel allocated space.
+    const timeToPixelMatrix = useMemo(() => get1dTimeToPixelMatrix(pixelsPerSecond, startTimeSec),
+        [pixelsPerSecond, startTimeSec])
+
+    const series = useMemo(() => {
+        const series: {dimensionIndex: number, times: number[], values: number[]}[] = []
+        for (let dimensionIndex=0; dimensionIndex<data.dimensionLabels.length; dimensionIndex++) {
+            const filteredTimes = data.timestamps.filter(t => (startTimeSec <= t) && (t <= endTimeSec))
+            const filteredValues = data.timestamps.map((t, ii) => (data.positions[ii][dimensionIndex])).filter((a, ii) => (startTimeSec <= data.timestamps[ii]) && (data.timestamps[ii] <= endTimeSec))
+            series.push({
+                dimensionIndex,
+                times: filteredTimes,
+                values: filteredValues
+            })
+        }
+        return series
+    }, [data.timestamps, startTimeSec, endTimeSec, data.dimensionLabels, data.positions])
+
+    const valueRange = useMemo(() => {
+        const yMin = Math.min(0, min(series.map(S => (min(S.values)))))
+        const yMax = Math.max(0, max(series.map(S => (max(S.values)))))
+        return {yMin, yMax}
+    }, [series])
+
+    const panels: TimeScrollViewPanel<PanelProps>[] = useMemo(() => {
+        const yMap = ((y: number) => (
```

Comment:
Wondering if this means now is the time to get around to implementing that two-dimensional transformation matrix function. (hmm)

Reply:
Actually--now I look at this in the gui, I'm not actually sure what's being plotted here. Time is on the x-axis, and the y-axis is the y-value of the animal's position? (Or some function of the x, y position?)

Also, looking at the output, I think we might want to think about ways to smooth or interpolate this data some.

Reply:
The y value is the value of which ever coordinate we're plotting. And it's possible to plot more than one coordinate on the same graph. For example, here's the x- and y- coordinates plotted on the same graph:

![image](https://user-images.githubusercontent.com/3679296/140984910-c251a2f2-9ad7-4536-a924-3f7f9c326ab0.png)

Reply:
Ah, okay. As a non-expert, I don't have much intuition for what this graph is telling me, but if it makes sense to people working in the field, that's what matters.

Is the oscillation pattern a product of the data--like, the animal is actually rocking back and forth in some way? Or is it an artifact of sampling rates or something?

Reply:
The animal is running back and forth along a track. The timescale is such that each oscillation could be a minute or more.

Reply:
Oh, okay. With that scale in mind it makes a lot more sense.